src/api/erc20/Erc20Api.ts

```diff
@@ -9,15 +9,20 @@ import { toBN } from 'utils'
 
 import Web3 from 'web3'
 
-export interface ContractReadOnlyParams {
+interface BaseParams {
```

Comment:
nit: I'm ok with this, but don't you prefer a name that would tell you what's inside?

for example `TokenAddress`

The lines like this one would read easily:
`export type NameParams = TokenAddress`

Reply:
Would the same style as this be better?

```typescript
interface WithTxOptionalParams {
  txOptionalParams?: TxOptionalParams
}
```

Thus we'd have:

```typescript
interface WithTokenAddress {
  tokenAddress: string
}
```

Reply:
My concern with this is we might need to create one interface per parameter.
On the follow up PR (#395) I'm adding `tokenId` to the BaseParams interface.

The main reason for creating this base interface was so I could add stuff to it that is available to all methods without lots of changes.